cdk/lib/core/core-service.ts

```diff
@@ -0,0 +1,101 @@
+import path = require('path');
+import { CfnService } from 'aws-cdk-lib/aws-apprunner';
+import { DockerImageAsset } from 'aws-cdk-lib/aws-ecr-assets';
+import {
+  ManagedPolicy,
+  PolicyStatement,
+  Role,
+  ServicePrincipal,
+} from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+
+export interface CoreServiceProps {
+  readonly databaseTableArn: string;
+  readonly databaseTableName: string;
+  readonly userPoolClientId: string;
+  readonly userPoolId: string;
+}
+
+export class CoreService extends Construct {
+  readonly databaseTableArn: string;
+  readonly databaseTableName: string;
+  readonly userPoolId: string;
+  readonly userPoolClientId: string;
+
+  constructor(scope: Construct, id: string, props: CoreServiceProps) {
+    super(scope, id);
+
+    ({
+      databaseTableArn: this.databaseTableArn,
+      databaseTableName: this.databaseTableName,
+      userPoolClientId: this.userPoolClientId,
+      userPoolId: this.userPoolId,
+    } = props);
```

Comment:
Nit: does this need parens?

Reply:
The parentheses are required, from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment):

> The parentheses ( ... ) around the assignment statement are required when using object literal destructuring assignment without a declaration.

the other way to do this is to first destructure into const vars and then assign to fields:

```
  const {
    databaseTableArn,
    databaseTableName,
    ...
  } = props;

  this.databaseTableArn = databaseTableArn;
  this.databaseTableName = databaseTableName;
  ...
```

but I prefer doing the destructuring assignment so it skipped the unnecessary const vars assignments